tests/test_helpers.py

```diff
@@ -515,6 +516,106 @@ def test_proxies_from_env_http_with_auth(mocker) -> None:
+
+
+@pytest.mark.parametrize(
+    ("proxies", "url_input", "expected_err_msg"),
+    (
+        (
+            {"no_proxy": ["aiohttp.io"], "NO_PROXY": ["aiohttp.io"]},
+            "http://aiohttp.io/path",
+            r"Proxying is disallowed for `'aiohttp.io'`",
+        ),
+        (
+            {
+                "http_proxy": ["http://example.com"],
+                "HTTP_PROXY": ["http://example.com"],
+            },
+            "https://aiohttp.io/path",
+            r"No proxies found for `https://aiohttp.io/path` in the env",
+        ),
+        (
+            {
+                "https_proxy": ["https://example.com"],
+                "HTTPS_PROXY": ["https://example.com"],
+            },
+            "http://aiohttp.io/path",
+            r"No proxies found for `http://aiohttp.io/path` in the env",
+        ),
+        (
+            {},
+            "https://aiohttp.io/path",
+            r"No proxies found for `https://aiohttp.io/path` in the env",
+        ),
+        (
+            {
+                "https_proxy": ["https://example.com"],
+                "HTTPS_PROXY": ["https://example.com"],
+            },
+            "",
+            r"No proxies found for `` in the env",
+        ),
+    ),
+    ids=(
+        "url_matches_the_no_proxy_list",
+        "url_scheme_does_not_match_http_proxy_list",
+        "url_scheme_does_not_match_https_proxy_list",
+        "no_proxies_are_set",
+        "url_is_empty",
+    ),
+)
+def test_get_env_proxy_for_url_negative(
+    monkeypatch, proxies, url_input, expected_err_msg
+) -> None:
+    for schema in getproxies_environment().keys():
+        monkeypatch.delenv(schema, False)
+
+    for proxy_type, proxy_list in proxies.items():
+        monkeypatch.setenv(proxy_type, ",".join(proxy_list))
+
+    url = URL(url_input)
+    with pytest.raises(LookupError, match=expected_err_msg):
+        helpers.get_env_proxy_for_url(url)
+
+
+@pytest.mark.parametrize(
+    ("proxies", "url_input", "expected_err_msg"),
+    (
+        (
+            {
+                "http_proxy": ["http://example.com"],
+                "HTTP_PROXY": ["http://example.com"],
+            },
+            "http://aiohttp.io/path",
+            r"No proxies found for `http://aiohttp.io/path` in the env",
+        ),
+        (
+            {
+                "https_proxy": ["http://example.com"],
+                "HTTPS_PROXY": ["http://example.com"],
+            },
+            "https://aiohttp.io/path",
+            r"No proxies found for `http://aiohttp.io/path` in the env",
+        ),
+    ),
+    ids=("url_scheme_match_http_proxy_list", "url_scheme_match_https_proxy_list"),
+)
+def test_get_env_proxy_for_url(
+    monkeypatch, proxies, url_input, expected_err_msg
+) -> None:
+    for schema in getproxies_environment().keys():
+        monkeypatch.delenv(schema, False)
+
+    for proxy_type, proxy_list in proxies.items():
+        monkeypatch.setenv(proxy_type, ",".join(proxy_list))
```

Comment:
Future note: normally I'd move this setup step into a pytest fixture and use "indirect" to bypass the params there (as in `@pytest.mark.parametrize(('proxy_env_vars', ...), ..., indirect=('proxy_env_vars', ))` + `@pytest.mark.usefixtures('proxy_env_vars')`).

It could look like:
```python
@pytest.fixture
def proxy_env_vars(request):
    for schema in getproxies_environment().keys():
        monkeypatch.delenv(schema, False)

    for var_name, var_value in request.param.items():
        monkeypatch.setenv(var_name, var_value)
```

Reply:
I like this. Done.